src/edu/ucsb/nceas/metacat/DocumentImpl.java

```diff
@@ -2555,60 +2406,58 @@ private static XMLReader initializeParser(DBConnection dbconn,
+                    throw new Exception("The schema for the document " + docid
+                                            + " can't be found in any place. So we can't validate"
+                                            + " the xml instance.");
                 }
-            } else if (ruleBase != null && ruleBase.equals(NONAMESPACESCHEMA)
-                    && needValidation) {
+            } else if (ruleBase != null && ruleBase.equals(NONAMESPACESCHEMA) && needValidation) {
                 //xmlss.doRefresh();
                 logMetacat.info("DocumentImpl.initalizeParser - Using General schema parser");
                 // turn on schema validation feature
                 parser.setFeature(VALIDATIONFEATURE, true);
                 parser.setFeature(NAMESPACEFEATURE, true);
                 //parser.setFeature(NAMESPACEPREFIXESFEATURE, true);
                 parser.setFeature(SCHEMAVALIDATIONFEATURE, true);
-                logMetacat.info("DocumentImpl.initalizeParser - Generic external no-namespace schema location: " + schemaLocation);              
+                logMetacat.info(
+                    "DocumentImpl.initalizeParser - Generic external no-namespace schema location: "
+                        + schemaLocation);
                 // Set external schemalocation.
-                if (schemaLocation != null
-                        && !(schemaLocation.trim()).equals("")) {
-                    parser.setProperty(EXTERNALNONAMESPACESCHEMALOCATIONPROPERTY,
-                            schemaLocation);
+                if (schemaLocation != null && !(schemaLocation.trim()).equals("")) {
+                    parser.setProperty(EXTERNALNONAMESPACESCHEMALOCATIONPROPERTY, schemaLocation);
                 } else {
-                    throw new Exception ("The schema for the document "+docid+" can't be found in any place. So we can't validate the xml instance.");
+                    throw new Exception("The schema for the document " + docid
+                                            + " can't be found in any place. So we can't validate"
+                + " the xml instance.");
                 }
-            } else if (ruleBase != null && ruleBase.equals(DTD)
-                    && needValidation) {
+            } else if (ruleBase != null && ruleBase.equals(DTD) && needValidation) {
                 logMetacat.info("DocumentImpl.initalizeParser - Using dtd parser");
                 // turn on dtd validaton feature
                 parser.setFeature(VALIDATIONFEATURE, true);
-                eresolver = new DBEntityResolver(dbconn,
-                        (DBSAXHandler) chandler, dtd);
+                eresolver = new DBEntityResolver(dbconn, chandler, dtd);
                 dtdhandler = new DBDTDHandler(dbconn);
-                parser.setEntityResolver((EntityResolver) eresolver);
-                parser.setDTDHandler((DTDHandler) dtdhandler);
```

Comment:
Did test without casting?

Reply:
`dtdhandler` is already of type `DTDHandler` - [see definition above](https://github.com/NCEAS/metacat/pull/1725/files#diff-bdee27d50eaffc92fec27fd02305f2554e85018ed989baa75b3bd3c608c34747R2384):

```java
            DTDHandler dtdhandler;
```
...so casting is unnecessary

Reply:
disregard - will revert changes and recommit without code cleanup